### PR TITLE
Sliding Cross Correlation improvement

### PR DESCRIPTION
**Pull request by Wosser1sProductions**
````
    + Re-added sliding cross correlation, but now starts from the index found 
        by the falling edge detector, which enables very fast cross correlation 
        and yields much more accurate results
    ~ Changed index_correction correlation percentage threshold 
        from 0.8 to 0.9 for accuracy improvements
--------
    ~ Changed prototypes to have const args where possible
    ~ Fixed missing multiplier in window size
--------
    ~ Renamed some variables
    + Added break in sliding_cross_correlate_upchirp if correlation percentage drops
    + Added DBGR macros to measure execution time of functions
````

### CODE REVIEW

File: lib/decoder_impl.cc [hunk 1]
```diff
@@ -315,9 +315,13 @@ namespace gr {
         float decoder_impl::sliding_norm_cross_correlate_upchirp(const float *samples, uint32_t window, uint32_t slide, int32_t *index) {
             (void) slide;
 
-            bool found_change = false;
-            uint32_t max = 0u, min;
-            const uint32_t coeff = 16u;
+            bool found_change     = false;
+            uint32_t max          = 0u, min;
+            
+            const uint32_t coeff  = (this->d_sf + this->d_sf + this->d_sf >> 1u);
```
Comment (rpp0) [inline lib/decoder_impl.cc]: Remove bitshift

File: lib/decoder_impl.cc [hunk 2]
```diff
@@ -331,23 +335,20 @@ namespace gr {
...
-            *index = max;
-
-            // Extra to allow falling edge on 99% of sample == Practically synced
-            // Should improve detection by not discarding almost synced samples, but doesn't..
-//            if ((max - coeff / 2) > (window * 0.98f)) {
-////                printf("ALREADY SYNCED\n");
-//                *index = 0;
-//                return 0.99f;
-//            }
+            max = std::max_element(samples + gr::lora::clamp((int)(max -  2u * coeff),  0, (int)window),
```
Comment (rpp0) [inline lib/decoder_impl.cc]: Naming of ```max``` and ```min``` should be changed to be less confusing: the variable ```max``` is for example reused for different purposes (as some value in the middle, as the maximum value of the symbol, and as the argmax of this value), making the code hard to read.

File: lib/decoder_impl.cc [hunk 3]
```diff
@@ -770,7 +769,7 @@ namespace gr {
                                                        this->d_samples_per_symbol / this->d_corr_decim_factor,
                                                        &index_correction);
 
-                        if (c > 0.8f) {
+                        if (c > 0.9f) {
```
Comment (rpp0) [inline lib/decoder_impl.cc]: This value is too high for the RTL-SDR. Either some filtering must be done or the correlation must be set to 0.8f

Comment (Wosser1sProductions) [inline lib/decoder_impl.cc]: Due to a missing factor 2 in the function call to `sliding_cross_correlate_upchirp`, this was indeed wrong. However, the new algorithm demonstrates much higher accuracy results with 0.9.

**Pull request merged**